packages/db/src/network/test/mockProvider.ts

```diff
@@ -1,27 +1,31 @@
 import { logger } from "@truffle/db/logger";
 const debug = logger("db:network:test:mockProvider");
 
-import type { Provider } from "web3/providers";
+import type {
+  Web3BaseProvider,
+  JsonRpcIdentifier,
+  JsonRpcRequest
+} from "web3-types";
 
 import { Batch, Model } from "test/arbitraries/networks";
 
 export const mockProvider = (options: {
   model: Model;
   batch: Batch;
-}): Provider => {
+}): Web3BaseProvider => {
   const { model, batch } = options;
 
   const { networkId, getBlockByNumber } = model.networks[batch.descendantIndex];
 
   return {
     send(payload, callback) {
-      const { jsonrpc, id, method, params } = payload;
+      const { jsonrpc, id, method, params } = payload as JsonRpcRequest<any>;
 
       switch (method) {
         case "eth_getBlockByNumber": {
           const [blockNumber] = params;
 
-          const height = parseInt(blockNumber);
+          const height = parseInt(blockNumber as string);
```

Comment:
This shouldn't have to be type casted, what type is it? (@spacesailor24)

Reply:
`blockNumber` is a number of strings. Either we can add a condition or typecast this value.